src/main/java/com/github/onsdigital/thetrain/storage/Transactions.java

```diff
@@ -110,14 +122,136 @@ public static void end(Transaction transaction) {
+        // the transaction threshold date/time.
+        for (String id : transactionsInFolder) {
+            Transaction transaction = null;
+            try {
+                transaction = Transactions.get(id);
+            } catch (IOException e) {
+                error().transactionID(id).exception(e).log("IOException when getting the transaction:" + id);
+                // Erroneous transactions also require archiving
+                id = (id.equals(null)) ? "null" : id;
+                Integer size = (transaction == null) ? 0 : transaction.errors().size();
+                // ToDo - create another folder for ErroneousTransactions, and associated Env Variable
+                transactionsToBeArchived.put(id, size);
+                continue;
+            }
+            //
+            // The STATUS of Transaction could indicate that it should be archived
+            //
+            if(archiveBasedOnStatus(transaction))
+            {
+                // This transaction should be archived
+                transactionsToBeArchived.put(id, transaction.errors().size());
+                // Increment counter reason for archiving transaction
+                tallyArchivalReason(transaction.getStatus());
+                // Move onto next transaction
+                continue;
+            }
+
+            //
+            // The StartDate or EndDate could indicate that it should be archived.
+            //
+            if (archiveBasedOnTimeThreshold(transaction, transactionThreshold)) {
+                // Transaction is older than allowable so store id, tally archival reason and move onto next transaction
+                transactionsToBeArchived.put(id, transaction.errors().size());
+                tallyArchivalReason("past threshold (" + transactionThreshold.toString() + ")");
+                continue;
+            }
+            // ToDo - Extract details of Transaction for displaying in Slack Message
+            // ToDo - THIRD CHECK - The datestamp of the newest Transaction folder/file
+        }
+        //
+        // Move the transactions to archive folder and send a Slack message
+        //
+        for (Map.Entry<String, Integer> entry : transactionsToBeArchived.entrySet()) {
+            String i = entry.getKey();
+            try {
+                FileUtils.moveToDirectory(new File(Paths.get(transactionStore.toString(), i).toAbsolutePath().toString()),
+                        new File(archivedTransactionStore.toAbsolutePath().toString()), true);
+            } catch (IOException e) {
+                error().transactionID(i).exception(e).log("IOException when moving the transaction to archive, where id=" + i);
+            }
+        }
+        try {
+            Slack.sendSlackArchivalReasons(reasonsForArchiving, "The-Train Initialisation Archival Complete (" + transactionsToBeArchived.size() + " archives)");
+        } catch (Exception e) {
```

Comment:
Is it possible to find out which type of exception is being caught, as it's better practice to catch specific exceptions than the generic one.